test/integration/discovery.go

```diff
@@ -306,13 +309,161 @@ var _ = Describe("Discover Cluster", func() {
 			entityDTOs, _, err := discoveryClient.DiscoverWithNewFramework("discovery-integration-test")
 			framework.ExpectNoError(err, "Failed completing discovery of test cluster")
 
+			podEntityDTO, nodeEntityDTO = getPodNodeDTO(entityDTOs, podFullName)
+
 			//Validation logic
-			if !validateBuyerSellerCommodity(entityDTOs, nodeNameForPod, podFullName, proto.CommodityDTO_VMPM_ACCESS, "foo in (bar)") {
+			if !validateBuyerSellerCommodity(podEntityDTO, nodeEntityDTO, nodeNameForPod, proto.CommodityDTO_VMPM_ACCESS, "foo in (bar)") {
 				framework.Failf("PV affinity is not honored")
 			}
 		})
 	})
 
+	Describe("discovering pod with pv having node with zone label", func() {
+		It("discovering pod with pv having node with zone label with default featureGate value true", func() {
+			var err error
+			var podFullName string
+			var nodeName string
+			var commodityZoneValue string
+			var commodityRegionValue string
+			var zoneNode *corev1.Node
+			var podEntityDTO *proto.EntityDTO
+			var nodeEntityDTO []*proto.EntityDTO
+			var zoneLabelKey = "topology.kubernetes.io/zone"
+			var regionLabelKey = "topology.kubernetes.io/region"
+			var zoneLabelValue = "us-west-1"
+			var regionLabelValue = "us-west-1"
+
+			pvc, err := createVolumeClaim(kubeClient, namespace, "standard")
+			if pvc == nil && err != nil {
+				framework.Failf("Failed to create PVC %s with the error %s", pvc, err)
+			}
+
+			dep, err := createDeployResource(kubeClient, depSingleContainerWithResources(namespace, pvc.Name, 1, true, false, false, ""))
+			framework.ExpectNoError(err, "Error creating test resources")
+
+			pod, err := getPodWithNamePrefix(kubeClient, dep.Name, namespace, "")
+			framework.ExpectNoError(err, "Error getting deployments pod")
+			// This should not happen. We should ideally get a pod.
+			if pod == nil {
+				framework.Failf("Failed to find a pod for deployment: %s", dep.Name)
+			}
+
+			//Label node with zone and region labels
+			nodeName = pod.Spec.NodeName
+			zoneNode, err = kubeClient.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
+			if err != nil {
+				framework.Failf("Failed to retrieve node with error %s", err)
+			}
+			zoneNode.Labels[zoneLabelKey] = zoneLabelValue //Label node with zone labels
+			zoneNode.Labels[regionLabelKey] = regionLabelValue
+			zoneNode, err = kubeClient.CoreV1().Nodes().Update(context.TODO(), zoneNode, metav1.UpdateOptions{})
```

Comment:
If we label the node here,  do we still need to label the node in the script `hack/create_kind_cluster.sh`?

Reply:
@kevinwangcn It is in a different test case. I was planning to change that implementation and test it out.